File: unified-workspace-sdk/sdk-typescript/examples/basic.ts
/**
 * Example: Basic Usage
 *
 * This example demonstrates basic sandbox and process operations.
 *
 * Run: npx ts-node basic.ts
 */

import { WorkspaceClient } from '../src';

async function main() {
  console.log('=== Workspace SDK Basic Example ===\n');

  const client = new WorkspaceClient({
    apiUrl: 'http://localhost:8080',
    timeout: 60000,
  });

  // 1. Create a sandbox
  console.log('1. Creating sandbox...');
  const sandbox = await client.sandbox.create({
    template: 'workspace-test:latest',
    name: 'example-sandbox',
    metadata: { purpose: 'demo' },
  });
  console.log(`   Created: ${sandbox.id} (state: ${sandbox.state})\n`);

  try {
    // 2. Run a simple command
    console.log('2. Running echo command...');
    let result = await client.process.run(sandbox.id, 'echo', {
      args: ['Hello', 'from', 'TypeScript', 'SDK!'],
    });
    console.log(`   Output: ${result.stdout}`);

    // 3. Run command with environment variables
    console.log('3. Running command with environment variables...');
    result = await client.process.run(sandbox.id, 'bash', {
      args: ['-c', 'echo "User: $USER, App: $APP_NAME"'],
      env: { USER: 'developer', APP_NAME: 'MyApp' },
    });
    console.log(`   Output: ${result.stdout}`);

    // 4. Write and read a file
    console.log('4. Writing and reading a file...');
    result = await client.process.run(sandbox.id, 'bash', {
      args: [
        '-c',
        `echo '{"name": "test", "version": "1.0.0"}' > /workspace/config.json && cat /workspace/config.json`,
      ],
    });
    console.log(`   File content: ${result.stdout}`);

    // 5. List workspace directory
    console.log('5. Listing workspace directory...');
    result = await client.process.run(sandbox.id, 'ls', {
      args: ['-la', '/workspace'],
    });
    console.log(`   Directory listing:\n${result.stdout}`);
  } finally {
    // 6. Cleanup
    console.log('\n6. Cleaning up...');
    await client.sandbox.delete(sandbox.id, true);
    console.log('   Done!');
  }
}

main().catch(console.error);

File: sdk-python/examples/basic.py
#!/usr/bin/env python3
"""
Example: Basic Usage

This example demonstrates basic sandbox and process operations
using the synchronous client.

Run: python basic.py
"""

from workspace_sdk import (
    WorkspaceClient,
    CreateSandboxParams,
    RunCommandOptions,
)


def main():
    print("=== Workspace SDK Basic Example ===\n")

    with WorkspaceClient("http://localhost:8080", timeout=60.0) as client:
        # 1. Create a sandbox
        print("1. Creating sandbox...")
        sandbox = client.sandbox.create(CreateSandboxParams(
            template="workspace-test:latest",
            name="example-sandbox",
            metadata={"purpose": "demo"},
        ))
        print(f"   Created: {sandbox.id} (state: {sandbox.state})\n")

        try:
            # 2. Run a simple command
            print("2. Running echo command...")
            result = client.process.run(
                sandbox.id,
                "echo",
                RunCommandOptions(args=["Hello", "from", "Python", "SDK!"])
            )
            print(f"   Output: {result.stdout}")

            # 3. Run command with environment variables
            print("3. Running command with environment variables...")
            result = client.process.run(
                sandbox.id,
                "bash",
                RunCommandOptions(
                    args=["-c", "echo \"User: $USER, App: $APP_NAME\""],
                    env={"USER": "developer", "APP_NAME": "MyApp"},
                )
            )
            print(f"   Output: {result.stdout}")

            # 4. Write and read a file
            print("4. Writing and reading a file...")
            result = client.process.run(
                sandbox.id,
                "bash",
                RunCommandOptions(args=["-c", """
                    echo '{"name": "test", "version": "1.0.0"}' > /workspace/config.json
                    cat /workspace/config.json
                """])
            )
            print(f"   File content: {result.stdout}")

            # 5. List workspace directory
            print("5. Listing workspace directory...")
            result = client.process.run(
                sandbox.id,
                "ls",
                RunCommandOptions(args=["-la", "/workspace"])
            )
            print(f"   Directory listing:\n{result.stdout}")

        finally:
            # 6. Cleanup
            print("\n6. Cleaning up...")
            client.sandbox.delete(sandbox.id, force=True)
            print("   Done!")


if __name__ == "__main__":
    main()
